src/fromager/wheels.py

```diff
@@ -308,3 +309,23 @@ def default_build_wheel(
             extra_environ=override_env,
             network_isolation=ctx.network_isolation,
         )
+
+
+def _add_build_tag(
+    req: Requirement,
+    wheel_filename: pathlib.Path,
+    build_tag: int = 0,
+) -> pathlib.Path:
+    cmd = [
+        "wheel",
+        "tags",
+        "--build",
+        str(build_tag),
+        "--remove",
+        str(wheel_filename),
+    ]
```

Comment:
`wheel pack` can also add a build tag. For efficiency, I recommend to fold this feature into `add_extra_metadata_to_wheel`. `wheel tag` unpacks the wheel file and repacks a new wheel file with a build tag. Every unpack and pack operation also validates or updates the SHA256 digest of every file. It's slow for large wheels like PyTorch.